scapy/layers/l2tp.py

```diff
@@ -16,14 +16,29 @@
 from scapy.layers.inet import UDP
 from scapy.layers.ppp import PPP
 
+_fields_present = ['res00', 'res01', 'res02', 'res03', 'priority', 'offset',
+                   'res06', 'sequence', 'res08', 'res09', 'length', 'type']
+
+
 class L2TP(Packet):
-    fields_desc = [ ShortEnumField("pkt_type",2,{2:"data"}),
-                    ShortField("len", None),
+    name = "L2TP"
+    fields_desc = [ 
+                    FlagsField("hdr", 0, len(_fields_present), _fields_present),
```

Comment:
Could you replace that with something like this (and remove the `_fields_present` global):
```python
    FlagsField("hdr", 0, 12, ['res00', 'res01', 'res02', 'res03', 'priority', 'offset', 'res06', 'sequence',
                                              'res08', 'res09', 'length', 'type']),
```